ethers-core/src/utils/mod.rs

```diff
@@ -456,53 +462,20 @@ fn estimate_priority_fee(rewards: Vec<Vec<U256>>) -> U256 {
     if rewards.len() == 1 {
         return rewards[0]
     }
-    // Sort the rewards as we will eventually take the median.
     rewards.sort();
+    let n = rewards.len();
+    let median =
+        if n % 2 == 0 { (rewards[n / 2 - 1] + rewards[n / 2]) / 2 } else { rewards[n / 2] };
 
-    // A copy of the same vector is created for convenience to calculate percentage change
-    // between subsequent fee values.
-    let mut rewards_copy = rewards.clone();
-    rewards_copy.rotate_left(1);
-
-    let mut percentage_change: Vec<I256> = rewards
-        .iter()
-        .zip(rewards_copy.iter())
-        .map(|(a, b)| {
-            let a = I256::try_from(*a).expect("priority fee overflow");
-            let b = I256::try_from(*b).expect("priority fee overflow");
-            ((b - a) * 100.into()) / a
-        })
-        .collect();
-    percentage_change.pop();
-
-    // Fetch the max of the percentage change, and that element's index.
-    let max_change = percentage_change.iter().max().unwrap();
-    let max_change_index = percentage_change.iter().position(|&c| c == *max_change).unwrap();
-
-    // If we encountered a big change in fees at a certain position, then consider only
-    // the values >= it.
-    let values = if *max_change >= EIP1559_FEE_ESTIMATION_THRESHOLD_MAX_CHANGE.into() &&
-        (max_change_index >= (rewards.len() / 2))
-    {
-        rewards[max_change_index..].to_vec()
-    } else {
-        rewards
-    };
-
-    // Return the median.
-    values[values.len() / 2]
```

Comment:
I think the median does change here after outliers are removed, since only low values are removed but high ones stay. So this returns different values than just returning the median of the sorted array with outliers in

Reply:
We actually remove low value twice: all zeros and after more than 50% of remaining ones if max change is high enough. Median statistics is robust to removal/changing no more 50% of entries, so, we certainly passed beyond the point where median statistics is robust.

The original implementation (with the removal) makes it more likely to land a transaction since it biases towards giving latest priority fees more weight.

It is not clear why alloy decided to go with bare median statics (commit comment does not have enough details).

Reply:
I restored the original version with different calculation of median (slightly better calculation).

Reply:
had a look at the code and I agree, definitely makes us more resilient to gas spikes. These two steps could be merged into a single one in case we ever need to optimize. 
```
    // Fetch the max of the percentage change, and that element's index.
    let max_change = percentage_change.iter().max().unwrap();
    let max_change_index = percentage_change.iter().position(|&c| c == *max_change).unwrap();
```